### PR TITLE
Added BYUN M415-6C smoke sensor

### PR DESCRIPTION
**Pull request by xyzroe**
https://item.taobao.com/item.htm?spm=a1z09.2.0.0.5c2e2e8d5AD1Ti&id=620637370020&_u=6209ep2lc6818f

### CODE REVIEW

File: devices.js [hunk 1]
```diff
@@ -15444,6 +15444,18 @@ const devices = [
...
     },
+
+    // BYUN
+    {
+        zigbeeModel: ['Windows switch  '],
+        model: 'M415-6C',
+        vendor: 'BYUN',
+        description: 'Smoke sensor',
+        supports: 'Smoke alarm',
+        fromZigbee: [fz.byun_smoke_on, fz.byun_smoke_off],
```
Comment (Koenkk) [inline devices.js:15455]: Can you try with `fromZigbee: [fz.ias_smoke_alarm_1]` instead here?

**Comment by Koenkk** [comment]
Note to self: after this merge https://github.com/Koenkk/zigbee2mqtt.io/pull/486

File: devices.js (hunk 1 above)
Comment (xyzroe) [reply devices.js:15455]: I conducted a test for 48 hours, tried low supply voltage, case tamper, test and real triggers: the sensor sends only two types of clusters `ssIasZone` and `pHMeasurement`, when triggered and restored, respectively.I tried with `ias_smoke_alarm_1`, this allows you to get a smoke alarm, but then it returns a lot of arguments not related to this device (enrolled, tamper, battery_low,  supervision_reports, restore_reports, trouble, ac_status).

It seems to me that information that is deceiving is not worth showing (battery, tamper so on), so I made my own converter. 
also the proposed converter allows to process the smoke sensor restore message.

Comment (xyzroe) [reply devices.js:15455]: I conducted a test, tried low supply voltage, case tamper, test and real triggers: the sensor sends only two types of clusters `ssIasZone` and `pHMeasurement`, when triggered and restored, respectively. [pastebin](https://pastebin.com/7M6Lzxif)
I tried with `ias_smoke_alarm_1`, this allows you to get a smoke alarm, but then it returns a lot of arguments not related to this device (enrolled, tamper, battery_low, supervision_reports, restore_reports, trouble, ac_status).
It seems to me that information that is deceiving is not worth showing (battery, tamper so on), so I made my own `fz.byun_smoke_on`
Also to process the smoke sensor restore message I made `fz.byun_smoke_off`

Comment (Koenkk) [reply devices.js:15455]: Thanks for looking into this.

**Comment by Koenkk** [comment]
Thanks!

**Comment by xyzroe** [comment]
Thanks for your work!

**Pull request merged**